### PR TITLE
[flang] Add FLANG_PARALLEL_COMPILE_JOBS option

### PR DESCRIPTION
**Pull request by tstellar**
This works the same way as LLVM_PARALLEL_COMPILE_JOBS except that it is specific to the flang source rather than for the whole project.

Configuring with -DFLANG_PARALLEL_COMPILE_JOBS=1 would mean that there would only ever be one flang source being compiled at a time.

Some of the flang sources require large amounts of memory to compile, so this option can be used to avoid OOM erros when compiling those files while still allowing the rest of the project to compile using the maximum number of jobs.

### CODE REVIEW

**Comment by tstellar** [comment]
If we know specifically which flang sources require a lot of memory to compile we could also add only those files to the pool, and keep the rest of the files in the default job pool.

**Review by joker-eph — APPROVED** [review]
Nice!

File: flang/CMakeLists.txt [hunk 1]
```diff
@@ -445,6 +445,18 @@ if (APPLE)
...
 
+# Set up job pools for flang.  Some of the flang sources take a lot of memory to
+# compile, so allow users to limit the number of parallel flang jobs.  This is
+# useful for building flang alongside several other projects since you can use
+# the maximum number of build jobs for the other projects while limiting the
+# number of flang compile jobs.
+#
+# We want this set to infinity by default
+set(FLANG_PARALLEL_COMPILE_JOBS 0 CACHE STRING
+  "The maxmimum number of concurrent compilation jobs (Ninja only)")
```
Comment (nikic) [inline flang/CMakeLists.txt:456]:
```suggestion
  "The maximum number of concurrent compilation jobs (Ninja only)")
```

**Review by kiranchandramohan — APPROVED** [review]
LG. Thanks for this change.

**Comment by kiranchandramohan** [comment]
One of the flang CI that tests builds for the flang runtime for offloading failed after this patch. 
https://lab.llvm.org/buildbot/#/builders/7/builds/197

**Comment by tstellar** [comment]
Updated PR with a fix for the runtime build: https://github.com/llvm/llvm-project/pull/127364

**Pull request merged**